### PR TITLE
Adding NCAR/geocat-viz package

### PR DESCRIPTION
**Pull request by pilotchute**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by pilotchute** [comment]
@NCAR/geocat comment here to confirm willingness to help maintain recipe.

**Comment by anissa111** [comment]
I am willing to be a maintainer

**Comment by michaelavs** [comment]
I am willing to be a maintainer

**Comment by hCraker** [comment]
Please add me to the list of maintainers. Thanks!

**Comment by pilotchute** [comment]
@erogluorhan Please comment :)

**Comment by erogluorhan** [comment]
I'd love to be a maintainer :)

**Comment by pilotchute** [comment]
@conda-forge-admin, please ping conda-forge/staged-recipes

File: recipes/geocat-viz/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    - scikit-learn
+    - xarray
+
+test:
+  requires:
+    - pip
+    - cmaps
+    - numpy
+    - matplotlib
+    - scikit-learn
```
Comment (dopplershift) [inline recipes/geocat-viz/meta.yaml:44]: Since most of these are in `run`, you don't need them. Also, unless you need a gui, better to use `matplotlib-base`, though I'm confused why it's only needed for tests?
```suggestion
    - matplotlib-base
```

File: recipes/geocat-viz/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  commands:
+    - pip check
+  source_files:
+    - src
+
+about:
+  home: https://geocat.ucar.edu/
+  summary: GeoCAT-viz is a collection of utility functions to make Python plotting with Matplotlib and Cartopy more like NCL (NCAR Command Language).
+  dev_url: https://github.com/NCAR/{{ name }}
+  doc_url: https://{{ name }}.readthedocs.io'
```
Comment (ocefpaf) [inline recipes/geocat-viz/meta.yaml:58]:
```suggestion
  doc_url: https://{{ name }}.readthedocs.io
```

File: recipes/geocat-viz/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    - geocat.viz.util
+  commands:
+    - pip check
+  source_files:
+    - src
+
+about:
+  home: https://geocat.ucar.edu/
+  summary: GeoCAT-viz is a collection of utility functions to make Python plotting with Matplotlib and Cartopy more like NCL (NCAR Command Language).
+  dev_url: https://github.com/NCAR/{{ name }}
```
Comment (ocefpaf) [inline recipes/geocat-viz/meta.yaml:57]: The use of the variable name here makes the recipe hard to read and links impossible to follow. Also, having a name var vs the actual name here does means you plan to change the name, which we hope you do not :smile:

TL;DR we only use jinja variables where they either help the bot or with thing we will change with every updating PR, like the version.

File: recipes/geocat-viz/meta.yaml (hunk 1 above)
Comment (pilotchute) [reply recipes/geocat-viz/meta.yaml:44]: I honestly don't know, I'm haven't developed in geocat-viz, but since I handled our team's other conda-forge recipes, I was chosen to get this package onto conda-forge. if it builds with those changes, I'm happy to make them.

File: recipes/geocat-viz/meta.yaml (hunk 2 above)
Comment (pilotchute) [reply recipes/geocat-viz/meta.yaml:58]: good catch, that's a typo.

File: recipes/geocat-viz/meta.yaml (hunk 3 above)
Comment (pilotchute) [reply recipes/geocat-viz/meta.yaml:57]: k, if you want a hardcoded name, we'll hardcode the name. 
we don't plan to change it.

File: recipes/geocat-viz/meta.yaml (hunk 1 above)
Comment (dopplershift) [reply recipes/geocat-viz/meta.yaml:44]: matplotlib is in  `requirements.txt` in the project's source repo, so it should also be deleted here and added above.

File: recipes/geocat-viz/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    - pip                                 # [build_platform != target_platform]
+    - setuptools                          # [build_platform != target_platform]
+    - cross-python_{{ target_platform }}  # [build_platform != target_platform]
+  host:
+    - python >=3.8
+    - pip
+    - setuptools
+  run:
+    - python >=3.8
+    - cartopy
```
Comment (dopplershift) [inline recipes/geocat-viz/meta.yaml:32]:
```suggestion
    - cartopy
    - matplotlib-base
```

File: recipes/geocat-viz/meta.yaml (hunk 2 above)
Comment (pilotchute) [reply recipes/geocat-viz/meta.yaml]: resolved

File: recipes/geocat-viz/meta.yaml (hunk 3 above)
Comment (pilotchute) [reply recipes/geocat-viz/meta.yaml]: resolved

File: recipes/geocat-viz/meta.yaml (hunk 4 above)
Comment (pilotchute) [reply recipes/geocat-viz/meta.yaml:31]: added matplotlib to match requirements.txt contents.
resolving.

File: recipes/geocat-viz/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+test:
+  requires:
+    - pip
+  imports:
+    - geocat
+    - geocat.viz
+  commands:
+    - pip check
+  source_files:
+    - src
```
Comment (carterbox) [inline recipes/geocat-viz/meta.yaml:48]:
```suggestion
```
I think these source files are not needed because you are no longer running the tests. If they are needed for the import tests to pass, then that is a problem because the end user will not have these files available during import.

**Review by carterbox — CHANGES_REQUESTED** [review]

File: recipes/geocat-viz/meta.yaml (hunk 5 above)
Comment (erogluorhan) [reply recipes/geocat-viz/meta.yaml:48]: @carterbox , as one of the maintainers listed here, I can remove that. Is that okay?

Comment (carterbox) [reply recipes/geocat-viz/meta.yaml:48]: Yes, that's fine. It's common for PRs to have multiple editors.

Comment (erogluorhan) [reply recipes/geocat-viz/meta.yaml]: Done, thanks!

**Comment by erogluorhan** [comment]
@conda-forge-admin, please ping conda-forge/staged-recipes

File: recipes/geocat-viz/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  noarch: python
+  number: 0
+  script: {{ PYTHON }} -m pip install --no-deps --ignore-installed -vv .
+
+requirements:
+  build:
+    - python                              # [build_platform != target_platform]
+    - pip                                 # [build_platform != target_platform]
+    - setuptools                          # [build_platform != target_platform]
+    - cross-python_{{ target_platform }}  # [build_platform != target_platform]
```
Comment (carterbox) [inline recipes/geocat-viz/meta.yaml:24]:
```suggestion
```
This package is `noarch: python` (no compiled code). Once you get a feedstock, it will be built *once* on linux-64, so these selectors for cross-compiling should not be needed.

**Review by carterbox — CHANGES_REQUESTED** [review]

File: recipes/geocat-viz/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+package:
+  name: geocat-viz
+  version: {{ version }}
+
+source:
+  url: https://github.com/NCAR/geocat-viz/archive/v{{ version }}.tar.gz   # For building from a specific tag's tarball
+  sha256: 021e1a7c8a9b6563fff130f58a50e79e8a664bb91d4efb4349811d9ee8f01f40
+  # git_tag: v{{ version }}   # For building from a specific tag
+  # git_rev: main # For building from main branch for debugging purposes
+  # git_url: https://github.com/NCAR/geocat-viz.git
```
Comment (carterbox) [inline recipes/geocat-viz/meta.yaml:12]:
```suggestion
```
REMINDER: You should always prefer building conda-forge packages from source tarballs.
